### PR TITLE
Feature/tec 3597 fe dynamic data

### PR DESCRIPTION
**Pull request by jesseeproductions**
The purpose of these changes to add filters and process the information from the backend to the front end. 
The modifications including enabling Pro to add in new fields to upgrade the List Widget to the Advanced List Widget.

_Ref:_ [TEC-3597](https://moderntribe.atlassian.net/browse/TEC-3597)
_Ref:_ [TEC-3620](https://moderntribe.atlassian.net/browse/TEC-3620)

Artifact: https://cloudup.com/cjmbbvnkLl2

Related:
https://github.com/moderntribe/tribe-common/pull/1460
https://github.com/moderntribe/events-pro/pull/1598

### CODE REVIEW

File: src/Tribe/Views/V2/Widgets/Widget_Abstract.php [hunk 1]
```diff
@@ -135,17 +129,18 @@ public function alter_context( Context $context, array $arguments = [] ) {
...
 	 *
 	 * @param array<string,mixed> $arguments Current set of arguments.
 	 * @param Context             $context   The request context.
 	 *
 	 * @return array<string,mixed> The translated widget arguments.
 	 */
 	protected function args_to_context( array $arguments, Context $context ) {
-		$context_args = [ 'widget' => true ];
+		$context_args = [
+			'widget' => true,
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Widgets/Widget_Abstract.php:141]: :warning: Array double arrow not aligned correctly; expected 7 space(s) between "'widget'" and double arrow, but found 1.

<sub>hash: 3a190549fd0e6d06198f98a142cf973f</sub>

File: src/Tribe/Views/V2/Widgets/Widget_List.php [hunk 2]
```diff
@@ -160,4 +162,25 @@ public function get_limit_options() {
 
 		return $options;
 	}
+
+	/**
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Widgets/Widget_List.php:166]: * :no_entry_sign: Doc comment for parameter "$arguments" missing
* :no_entry_sign: Doc comment for parameter "$context" missing

<sub>hash: 978817e56491e712d5623fa01f4dcf7e</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 2 error(s)
:warning: 1 warning(s)

File: src/Tribe/Views/V2/Widgets/Widget_List.php [hunk 3]
```diff
@@ -160,4 +164,31 @@ public function get_limit_options() {
...
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Widgets/Widget_List.php:168]: * :no_entry_sign: Doc comment for parameter "$arguments" missing
* :no_entry_sign: Doc comment for parameter "$context" missing

<sub>hash: 82f4604f7e1788c54dacbc033cfda794</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 2 error(s)

File: src/Tribe/Views/V2/Views/Widgets/Widget_List_View.php [hunk 4]
```diff
@@ -69,20 +69,34 @@ protected function setup_repository_args( Context $context = null ) {
...
+
+		/**
+		 * Applies a filter to the template vars for the List Widget.
+		 *
+		 * @since TBD
+		 *
+		 * @param array<string,mixed> $template_vars The updated instance of the widget.
+		 * @param \Tribe__Context $context   Tribe context used to setup the view.
+		 */
+		return apply_filters( 'tribe_events_view_v2_list_widget_template_vars', $template_vars, $this->context );
```
Comment (lucatume) [inline src/Tribe/Views/V2/Views/Widgets/Widget_List_View.php:100]: Isn't this value already filtered in the Abstract class in Common in the `filter_template_args` method?
I would remove this one as it seems to be redundant.

File: src/Tribe/Views/V2/Widgets/Widget_Abstract.php [hunk 5]
```diff
@@ -135,17 +129,18 @@ public function alter_context( Context $context, array $arguments = [] ) {
 	/**
 	 * Translates widget arguments to their Context argument counterpart.
 	 *
-	 * @todo update in TEC-3620 & TEC-3597
-	 *
-	 * @since 5.2.1
+	 * @since  5.2.1
```
Comment (lucatume) [inline src/Tribe/Views/V2/Widgets/Widget_Abstract.php:132]: Excess space here?

**Review by lucatume — CHANGES_REQUESTED** [review]
Some smaller adjustments.

File: src/Tribe/Views/V2/Views/Widgets/Widget_List_View.php (hunk 4 above)
Comment (jesseeproductions) [reply src/Tribe/Views/V2/Views/Widgets/Widget_List_View.php:100]: There is this filter in the View class: 
tribe_events_views_v2_view_{$this->slug}_template_vars

I removed the filter here and instead will use that in Pro to modify the template vars.

**Review by lucatume — APPROVED** [review]

**Pull request merged**